pkgs/development/go-modules/generic/default.nix

```diff
@@ -35,6 +40,31 @@ let
   removeExpr = refs: lib.flip lib.concatMapStrings refs (ref: ''
     | sed "s,${ref},$(echo "${ref}" | sed "s,$NIX_STORE/[^-]*,$NIX_STORE/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee,"),g" \
   '');
+
+  dep2src = goDep:
+    {
+      inherit (goDep) goPackagePath;
+      src = if goDep.fetch.type == "git" then
```

Comment:
is there a better way to make some pattern matching on fetch type? Or maybe I need to change json file structure to fit better to nix?